util/solver/executor/ci_local.py

```
@@ -0,0 +1,862 @@
+
+    def bind_problem(self, problem: ParsedProblem, bind_output: str | Path) -> None:
+        checkpoint = self._selected_checkpoint(problem)
+        cmd = self._build_gem5_command(problem, checkpoint, bind_output=bind_output)
+        bind_output = Path(bind_output)
+        bind_output.parent.mkdir(parents=True, exist_ok=True)
+        bind_log = bind_output.with_suffix(".log")
+        self._log(
+            f"binding {problem.name} with representative checkpoint {checkpoint}"
+        )
+        with bind_log.open("w", encoding="utf-8") as handle:
+            subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)